service/scaladsl/client/src/main/scala/com/lightbend/lagom/scaladsl/client/ServiceClient.scala

```diff
@@ -109,7 +110,7 @@ trait ServiceResolver {
 /**
  * The Lagom service client components.
  */
-trait LagomServiceClientComponents extends TopicFactoryProvider {
+trait LagomServiceClientComponents extends TopicFactoryProvider { self: LagomConfigComponent =>
```

Comment:
Nice trick!

It's not totally safe, but I guess we will be fine with this approach.
The self-type is probably not removing the binary incompatibility issue of `LagomServiceClientComponents`. It's probably because MiMa is not able to detect it for some reason (maybe a bug in MiMa).

I don't expect users to use `LagomServiceClientComponents` alone, but through `LagomClientApplication`. And the issue is being solved there when `LagomConfigComponent` is mixed in.

Reply:
I think binary compatibility is less critical for these components classes anyway.

Reply:
Yes, I went through the same kind conclusion. MiMa is happy with it but it should probably not...and at the same time, that's not an big issue in this case.